hack/prow/run_tests.py

```diff
@@ -0,0 +1,140 @@
+def upload_results(outdir, test, buildnum, bucket):
+  """ push the contents of gcs_out/* into bucket/test/logs/buildnum"""
+  classname = os.path.basename(test).split('.')[0]
+  args = shlex.split("gsutil cp -R gcs_out/ gs://%s/logs/%s/%s" % (bucket, classname, buildnum))
+  p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+  for line in p.stdout:
+    print line
+
+def gather_artifacts():
+  """ gather any named or default artifacts into artifacts/ """
+  return
+
+def run_tests(test, build_log, exit_status, started, finished, test_results):
+  """ execute the test, grab the start time, finish time, build logs and exit status
+      Pull test results and important information out of the build log 
+      test results format should be:
+      === RUN   TestFunctional/Mounting
+      --- PASS: TestFunctional (42.87s)
+          --- PASS: TestFunctional/Status (2.33s)
+          --- FAIL: SOMETESTSUITE/TESTNAME (seconds)
+  """
+  classname = os.path.basename(test).split('.')[0]
+  build_log_file = open(build_log, 'w')
+  args = shlex.split('bash -x %s' % test)
+  p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+  for line in p.stdout:
+    build_log_file.write(line)
+    print line.rstrip()
+    if '--- PASS' in line:
+      match = re.match('.*--- PASS: ([^ ]+) \(([0-9.]+)s\)', line)
+      (name, seconds) = match.group(1, 2)
+      test_results.append({"name":name, "classname":classname, "time":seconds, "status":"PASS"})
+    if '--- FAIL' in line:
+      match = re.match('.*--- FAIL: ([^ ]+) \(([0-9.]+)s\)', line)
+      (name, seconds) = match.group(1, 2)
+      test_results.append({"name":name, "classname":classname, "time":seconds, "status":"FAIL"})
+  build_log_file.close()
+  return
+
+def main(argv):
+
+  parser = argparse.ArgumentParser(description='Run tests and upload results to GCS bucket', usage='./run_tests.py --test path/to/test.sh')
+  parser.add_argument('--test', required=True, help='give full path to test script you want to run')
+  parser.add_argument('--buildnum', required=True, help='buildnumber for uploading to GCS')
+  parser.add_argument('--bucket', default="k8s-minikube-prow", help='Name of the GCS bucket to upload to.  Default: k8s-minkube-prow')
+  parser.add_argument('--outdir', default="gcs_out", help='Path of the directory to store all results, artifacts, and logs')
+  parser.add_argument('--artifact', help='SRCPATH:TARGETPATH use to specify a file that needs to be uploaded into GCS bucket')
+  args = parser.parse_args()
+
+  if not os.path.exists(args.outdir):
+    os.makedirs(args.outdir)
+    os.makedirs('%s/artifacts' % args.outdir)
+
+  build_log="%s/build_log.txt" % (args.outdir)
+  exit_status=""
+  started={"timestamp":calendar.timegm(time.gmtime())}
+  finished={}
+  test_results=[]
+
+  run_tests(args.test, build_log, exit_status, started, finished, test_results)
+
+  finished['timestamp'] = calendar.timegm(time.gmtime())
+  if exit_status != "0":
```

Comment:
Can you add a comment here explaining what's going on? Normally I would expect 0 to be success.